File: app/src/main/java/bj/discogsbrowser/model/common/RecyclerViewModel.java
package bj.discogsbrowser.model.common;

/**
 * Created by Josh Laird on 19/04/2017.
 */

public interface RecyclerViewModel
{
    String getTitle();

    String getSubtitle();

    String getThumb();

    String getType();

    // Has to be a String as Order IDs have hyphens
    String getId();
}

File: app/src/main/java/bj/discogsbrowser/model/common/RecyclerViewModel.kt
package bj.discogsbrowser.model.common

/**
 * Created by Josh Laird on 19/05/2017.
 */
interface RecyclerViewModel {
    fun getTitle(): String
    fun getSubtitle(): String
    fun getThumb(): String
    fun getType(): String
    fun getId(): String
}